command/server.go

```diff
@@ -1760,6 +1760,32 @@ func (c *ServerCommand) Run(args []string) int {
 		case <-c.SigUSR2Ch:
 			logWriter := c.logger.StandardWriter(&hclog.StandardLoggerOptions{})
 			pprof.Lookup("goroutine").WriteTo(logWriter, 2)
+
+			if os.Getenv("VAULT_WRITE_STACKTRACE_TO_FILE") != "" {
+				c.logger.Info("Writing stacktrace to file..")
+
+				dir, err := os.MkdirTemp("", "vault-stacktrace")
+				if err != nil {
+					c.logger.Error("Could not create temporary directory for stacktrace", err)
+					continue
+				}
+
+				f, err := os.Create(fmt.Sprintf("%s/%s", dir, "stacktrace.log"))
+				if err != nil {
+					c.logger.Error("Could not create temporary file for stacktrace", err)
+					continue
+				}
+				defer f.Close()
+
+				if err := pprof.Lookup("goroutine").WriteTo(f, 2); err != nil {
+					f.Close()
```

Comment:
Is this necessary given the `defer f.Close()` on line 1778?

Reply:
Ah, I see. So really, it was the defer that was out of place. Thanks for clarifying!